### ISSUE TITLE
[Bug]: [benchmark] Proxy memory grows over 120G in 48 hours of concurrency testing

### ISSUE THREAD

**Issue by elstic** [milvus-io/milvus#28367]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.3-20231110-f2e972ab
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):  pulsar  
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

argo task : fouramf-lwx4l 
case: test_concurrent_locust_100m_ivf_sq8_ddl_dql_filter_replica2_cluster
concurrent test time: 48h 

server:
```
fouramf-lwx4l-64-8132-etcd-0                                      1/1     Running     0               6m15s   10.104.17.145   4am-node23   <none>           <none>
fouramf-lwx4l-64-8132-etcd-1                                      1/1     Running     0               6m15s   10.104.21.45    4am-node24   <none>           <none>
fouramf-lwx4l-64-8132-etcd-2                                      1/1     Running     0               6m15s   10.104.19.204   4am-node28   <none>           <none>
fouramf-lwx4l-64-8132-milvus-datacoord-77f565b97d-vj72m           1/1     Running     0               6m15s   10.104.1.164    4am-node10   <none>           <none>
fouramf-lwx4l-64-8132-milvus-datanode-85b884cf8d-w6xmm            1/1     Running     1 (2m7s ago)    6m15s   10.104.6.230    4am-node13   <none>           <none>
fouramf-lwx4l-64-8132-milvus-indexcoord-7694894b5-x4b5b           1/1     Running     0               6m15s   10.104.13.230   4am-node16   <none>           <none>
fouramf-lwx4l-64-8132-milvus-indexnode-74c45fdf4b-6mjj8           1/1     Running     0               6m15s   10.104.1.165    4am-node10   <none>           <none>
fouramf-lwx4l-64-8132-milvus-proxy-784dfb6567-l2lc8               1/1     Running     1 (2m ago)      6m15s   10.104.13.233   4am-node16   <none>           <none>
fouramf-lwx4l-64-8132-milvus-querycoord-86d468f9bb-rqm67          1/1     Running     1 (2m6s ago)    6m15s   10.104.1.166    4am-node10   <none>           <none>
fouramf-lwx4l-64-8132-milvus-querynode-58bb495698-9w626           1/1     Running     0               6m15s   10.104.5.27     4am-node12   <none>           <none>
fouramf-lwx4l-64-8132-milvus-querynode-58bb495698-k28gf           1/1     Running     0               6m15s   10.104.13.229   4am-node16   <none>           <none>
fouramf-lwx4l-64-8132-milvus-rootcoord-5db49c4fd-xplxc            1/1     Running     1 (2m13s ago)   6m15s   10.104.5.28     4am-node12   <none>           <none>
fouramf-lwx4l-64-8132-minio-0                                     1/1     Running     0               6m15s   10.104.19.202   4am-node28   <none>           <none>
fouramf-lwx4l-64-8132-minio-1                                     1/1     Running     0               6m15s   10.104.21.38    4am-node24   <none>           <none>
fouramf-lwx4l-64-8132-minio-2                                     1/1     Running     0               6m15s   10.104.20.202   4am-node22   <none>           <none>
fouramf-lwx4l-64-8132-minio-3                                     1/1     Running     0               6m15s   10.104.17.150   4am-node23   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-bookie-0                             1/1     Running     0               6m15s   10.104.19.199   4am-node28   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-bookie-1                             1/1     Running     0               6m15s   10.104.17.146   4am-node23   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-bookie-2                             1/1     Running     0               6m15s   10.104.21.47    4am-node24   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-bookie-init-whkql                    0/1     Completed   0               6m15s   10.104.5.31     4am-node12   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-broker-0                             1/1     Running     0               6m15s   10.104.13.235   4am-node16   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-proxy-0                              1/1     Running     0               6m15s   10.104.13.231   4am-node16   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-pulsar-init-dzdpr                    0/1     Completed   0               6m15s   10.104.5.29     4am-node12   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-recovery-0                           1/1     Running     0               6m15s   10.104.5.30     4am-node12   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-zookeeper-0                          1/1     Running     0               6m15s   10.104.19.201   4am-node28   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-zookeeper-1                          1/1     Running     0               5m31s   10.104.21.49    4am-node24   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-zookeeper-2                          1/1     Running     0               4m54s   10.104.20.204   4am-node22   <none>           <none> (base.py:257)
[2023-11-12 15:16:40,055 -  INFO - fouram]: [Cmd Exe]  kubectl get pods  -n qa-milvus  -o wide | grep -E 'STATUS|fouramf-lwx4l-64-8132-milvus|fouramf-lwx4l-64-8132-minio|fouramf-lwx4l-64-8132-etcd|fouramf-lwx4l-64-8132-pulsar|fouramf-lwx4l-64-8132-kafka'  (util_cmd.py:14)
[2023-11-12 15:16:49,217 -  INFO - fouram]: [CliClient] pod details of release(fouramf-lwx4l-64-8132): 
 I1112 15:16:41.309304     539 request.go:665] Waited for 1.152705889s due to client-side throttling, not priority and fairness, request: GET:https://kubernetes.default.svc.cluster.local/apis/pingcap.com/v1alpha1?timeout=32s
NAME                                                              READY   STATUS        RESTARTS        AGE     IP              NODE         NOMINATED NODE   READINESS GATES
fouramf-lwx4l-64-8132-etcd-0                                      1/1     Running       0               2d3h    10.104.17.145   4am-node23   <none>           <none>
fouramf-lwx4l-64-8132-etcd-1                                      1/1     Running       0               2d3h    10.104.21.45    4am-node24   <none>           <none>
fouramf-lwx4l-64-8132-etcd-2                                      1/1     Running       0               2d3h    10.104.19.204   4am-node28   <none>           <none>
fouramf-lwx4l-64-8132-milvus-datacoord-77f565b97d-vj72m           1/1     Running       0               2d3h    10.104.1.164    4am-node10   <none>           <none>
fouramf-lwx4l-64-8132-milvus-datanode-85b884cf8d-w6xmm            1/1     Running       9 (12m ago)     2d3h    10.104.6.230    4am-node13   <none>           <none>
fouramf-lwx4l-64-8132-milvus-indexcoord-7694894b5-x4b5b           1/1     Running       0               2d3h    10.104.13.230   4am-node16   <none>           <none>
fouramf-lwx4l-64-8132-milvus-indexnode-74c45fdf4b-6mjj8           1/1     Running       0               2d3h    10.104.1.165    4am-node10   <none>           <none>
fouramf-lwx4l-64-8132-milvus-proxy-784dfb6567-l2lc8               1/1     Running       1 (2d3h ago)    2d3h    10.104.13.233   4am-node16   <none>           <none>
fouramf-lwx4l-64-8132-milvus-querycoord-86d468f9bb-rqm67          1/1     Running       1 (2d3h ago)    2d3h    10.104.1.166    4am-node10   <none>           <none>
fouramf-lwx4l-64-8132-milvus-querynode-58bb495698-9w626           1/1     Running       0               2d3h    10.104.5.27     4am-node12   <none>           <none>
fouramf-lwx4l-64-8132-milvus-querynode-58bb495698-k28gf           1/1     Running       0               2d3h    10.104.13.229   4am-node16   <none>           <none>
fouramf-lwx4l-64-8132-milvus-rootcoord-5db49c4fd-xplxc            1/1     Running       1 (2d3h ago)    2d3h    10.104.5.28     4am-node12   <none>           <none>
fouramf-lwx4l-64-8132-minio-0                                     1/1     Running       0               2d3h    10.104.19.202   4am-node28   <none>           <none>
fouramf-lwx4l-64-8132-minio-1                                     1/1     Running       0               2d3h    10.104.21.38    4am-node24   <none>           <none>
fouramf-lwx4l-64-8132-minio-2                                     1/1     Running       0               2d3h    10.104.20.202   4am-node22   <none>           <none>
fouramf-lwx4l-64-8132-minio-3                                     1/1     Running       0               2d3h    10.104.17.150   4am-node23   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-bookie-0                             1/1     Running       0               2d3h    10.104.19.199   4am-node28   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-bookie-1                             1/1     Running       0               2d3h    10.104.17.146   4am-node23   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-bookie-2                             1/1     Running       0               2d3h    10.104.21.47    4am-node24   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-bookie-init-whkql                    0/1     Completed     0               2d3h    10.104.5.31     4am-node12   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-broker-0                             1/1     Running       0               2d3h    10.104.13.235   4am-node16   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-proxy-0                              1/1     Running       0               2d3h    10.104.13.231   4am-node16   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-pulsar-init-dzdpr                    0/1     Completed     0               2d3h    10.104.5.29     4am-node12   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-recovery-0                           1/1     Running       0               2d3h    10.104.5.30     4am-node12   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-zookeeper-0                          1/1     Running       0               2d3h    10.104.19.201   4am-node28   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-zookeeper-1                          1/1     Running       0               2d3h    10.104.21.49    4am-node24   <none>           <none>
fouramf-lwx4l-64-8132-pulsar-zookeeper-2                          1/1     Running       0               2d3h    10.104.20.204   4am-node22   <none>           <none>
```
</b>
The datanode restart is another known issue that comes up here is the proxy memory growth validation. 

![image](https://github.com/milvus-io/milvus/assets/48523564/0651779f-7758-41c6-86d3-aa754b6c511d)

</b></b>
Normal proxy memory usage is less than 1 gigabyte.
![image](https://github.com/milvus-io/milvus/assets/48523564/86aec9ef-97db-42b7-97f7-1f8606f12697)

client pod : fouramf-lwx4l-3635258141

### Expected Behavior

proxy memory does not grow

### Steps To Reproduce

```markdown
1. create a collection or use an existing collection  
  2. build an IVF_SQ8 index on the vector column
  3. insert 100 million vectors
  4. flush collection
  5. build index on vector column with the same parameters  
  6. count the total number of rows
  7. load collection
  8. execute concurrent search, query,load,scene_test 
    (scene_test steps: 
       1） Create a collection 2） Insert 3000 pieces of data  3） flush collection 
       4） Create an index  5） drop collection)
  9. step 8 lasts 48h
```

### Anything else?

Our client side operation has a concurrency of 20, and there are some write forbidden errors in the scene_test phase.
Because the error in the scene_test phase will cause the 'drop collection' step not to be executed and exit directly. So after the test is completed, there is a lot of collection stacked up 
![image](https://github.com/milvus-io/milvus/assets/48523564/808da5b9-8be8-40cb-9841-d5e996900eaa)

**Comment by xiaofan-luan** [comment]
do we have pprof?

**Comment by elstic** [comment]
> do we have pprof?

We have reproduced this issue. I have reduced the amount of data and execution time, and it appears that the proxy memory is rising and there is a retention site.

server.
```
fouramf-t2wg2-57-6083-etcd-0                                      1/1     Running     0               9h      10.104.24.191   4am-node29   <none>           <none>
fouramf-t2wg2-57-6083-etcd-1                                      1/1     Running     0               9h      10.104.15.38    4am-node20   <none>           <none>
fouramf-t2wg2-57-6083-etcd-2                                      1/1     Running     0               9h      10.104.21.32    4am-node24   <none>           <none>
fouramf-t2wg2-57-6083-milvus-datacoord-5884474688-xnfn2           1/1     Running     0               9h      10.104.21.14    4am-node24   <none>           <none>
fouramf-t2wg2-57-6083-milvus-datanode-6bd97b66db-hvz5p            1/1     Running     1 (9h ago)      9h      10.104.20.112   4am-node22   <none>           <none>
fouramf-t2wg2-57-6083-milvus-indexcoord-7d4fc76bf5-lt97x          1/1     Running     0               9h      10.104.13.252   4am-node16   <none>           <none>
fouramf-t2wg2-57-6083-milvus-indexnode-6f74b4dd48-jvrbn           1/1     Running     0               9h      10.104.14.84    4am-node18   <none>           <none>
fouramf-t2wg2-57-6083-milvus-proxy-f57777f69-59sz8                1/1     Running     1 (9h ago)      9h      10.104.21.11    4am-node24   <none>           <none>
fouramf-t2wg2-57-6083-milvus-querycoord-544956499b-z26jh          1/1     Running     1 (9h ago)      9h      10.104.20.111   4am-node22   <none>           <none>
fouramf-t2wg2-57-6083-milvus-querynode-86cffb55f6-kg4mx           1/1     Running     0               9h      10.104.24.185   4am-node29   <none>           <none>
fouramf-t2wg2-57-6083-milvus-querynode-86cffb55f6-px89f           1/1     Running     0               9h      10.104.13.2     4am-node16   <none>           <none>
fouramf-t2wg2-57-6083-milvus-rootcoord-f88675f88-k5sg7            1/1     Running     1 (9h ago)      9h      10.104.13.251   4am-node16   <none>           <none>
fouramf-t2wg2-57-6083-minio-0                                     1/1     Running     0               9h      10.104.21.26    4am-node24   <none>           <none>
fouramf-t2wg2-57-6083-minio-1                                     1/1     Running     0               9h      10.104.24.190   4am-node29   <none>           <none>
fouramf-t2wg2-57-6083-minio-2                                     1/1     Running     0               9h      10.104.19.219   4am-node28   <none>           <none>
fouramf-t2wg2-57-6083-minio-3                                     1/1     Running     0               9h      10.104.20.130   4am-node22   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-bookie-0                             1/1     Running     0               9h      10.104.21.30    4am-node24   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-bookie-1                             1/1     Running     0               9h      10.104.20.124   4am-node22   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-bookie-2                             1/1     Running     0               9h      10.104.19.221   4am-node28   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-bookie-init-b65s2                    0/1     Completed   0               9h      10.104.24.183   4am-node29   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-broker-0                             1/1     Running     0               9h      10.104.24.184   4am-node29   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-proxy-0                              1/1     Running     0               9h      10.104.21.13    4am-node24   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-pulsar-init-nv2xb                    0/1     Completed   0               9h      10.104.14.83    4am-node18   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-recovery-0                           1/1     Running     0               9h      10.104.13.254   4am-node16   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-zookeeper-0                          1/1     Running     0               9h      10.104.19.218   4am-node28   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-zookeeper-1                          1/1     Running     0               9h      10.104.15.42    4am-node20   <none>           <none>
fouramf-t2wg2-57-6083-pulsar-zookeeper-2                          1/1     Running     0               9h      10.104.16.87    4am-node21   <none>           <none>
```

**Comment by xiaofan-luan** [comment]
> > do we have pprof?
> 
> We have reproduced this issue. I have reduced the amount of data and execution time, and it appears that the proxy memory is rising and there is a retention site.The following instance is observable pprof.
> 
> server.
> 
> ```
> fouramf-t2wg2-57-6083-etcd-0                                      1/1     Running     0               9h      10.104.24.191   4am-node29   <none>           <none>
> fouramf-t2wg2-57-6083-etcd-1                                      1/1     Running     0               9h      10.104.15.38    4am-node20   <none>           <none>
> fouramf-t2wg2-57-6083-etcd-2                                      1/1     Running     0               9h      10.104.21.32    4am-node24   <none>           <none>
> fouramf-t2wg2-57-6083-milvus-datacoord-5884474688-xnfn2           1/1     Running     0               9h      10.104.21.14    4am-node24   <none>           <none>
> fouramf-t2wg2-57-6083-milvus-datanode-6bd97b66db-hvz5p            1/1     Running     1 (9h ago)      9h      10.104.20.112   4am-node22   <none>           <none>
> fouramf-t2wg2-57-6083-milvus-indexcoord-7d4fc76bf5-lt97x          1/1     Running     0               9h      10.104.13.252   4am-node16   <none>           <none>
> fouramf-t2wg2-57-6083-milvus-indexnode-6f74b4dd48-jvrbn           1/1     Running     0               9h      10.104.14.84    4am-node18   <none>           <none>
> fouramf-t2wg2-57-6083-milvus-proxy-f57777f69-59sz8                1/1     Running     1 (9h ago)      9h      10.104.21.11    4am-node24   <none>           <none>
> fouramf-t2wg2-57-6083-milvus-querycoord-544956499b-z26jh          1/1     Running     1 (9h ago)      9h      10.104.20.111   4am-node22   <none>           <none>
> fouramf-t2wg2-57-6083-milvus-querynode-86cffb55f6-kg4mx           1/1     Running     0               9h      10.104.24.185   4am-node29   <none>           <none>
> fouramf-t2wg2-57-6083-milvus-querynode-86cffb55f6-px89f           1/1     Running     0               9h      10.104.13.2     4am-node16   <none>           <none>
> fouramf-t2wg2-57-6083-milvus-rootcoord-f88675f88-k5sg7            1/1     Running     1 (9h ago)      9h      10.104.13.251   4am-node16   <none>           <none>
> fouramf-t2wg2-57-6083-minio-0                                     1/1     Running     0               9h      10.104.21.26    4am-node24   <none>           <none>
> fouramf-t2wg2-57-6083-minio-1                                     1/1     Running     0               9h      10.104.24.190   4am-node29   <none>           <none>
> fouramf-t2wg2-57-6083-minio-2                                     1/1     Running     0               9h      10.104.19.219   4am-node28   <none>           <none>
> fouramf-t2wg2-57-6083-minio-3                                     1/1     Running     0               9h      10.104.20.130   4am-node22   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-bookie-0                             1/1     Running     0               9h      10.104.21.30    4am-node24   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-bookie-1                             1/1     Running     0               9h      10.104.20.124   4am-node22   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-bookie-2                             1/1     Running     0               9h      10.104.19.221   4am-node28   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-bookie-init-b65s2                    0/1     Completed   0               9h      10.104.24.183   4am-node29   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-broker-0                             1/1     Running     0               9h      10.104.24.184   4am-node29   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-proxy-0                              1/1     Running     0               9h      10.104.21.13    4am-node24   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-pulsar-init-nv2xb                    0/1     Completed   0               9h      10.104.14.83    4am-node18   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-recovery-0                           1/1     Running     0               9h      10.104.13.254   4am-node16   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-zookeeper-0                          1/1     Running     0               9h      10.104.19.218   4am-node28   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-zookeeper-1                          1/1     Running     0               9h      10.104.15.42    4am-node20   <none>           <none>
> fouramf-t2wg2-57-6083-pulsar-zookeeper-2                          1/1     Running     0               9h      10.104.16.87    4am-node21   <none>           <none>
> ```
> 
> memory usage: ![image](https://user-images.githubusercontent.com/48523564/282644338-92d64b11-713a-4433-9aa9-7ee593110a6b.png)

could we have one pprof when it reproduced

**Comment by elstic** [comment]
Yes, we have pprof 
 
[goroutine.zip](https://github.com/milvus-io/milvus/files/13344285/goroutine.zip)

**Comment by elstic** [comment]
This problem exists on both the 2.3 branch and the master branch, so if there is a fix, please cherry pick the fix to the master branch.

**Comment by smellthemoon** [comment]
![pprof001](https://github.com/milvus-io/milvus/assets/64083300/3f1199b0-ab92-49ff-b3e2-a6f898312d5f)
There is a goroutine internally to append data to a batch in pulsar. If you don't close the producer, the goroutine corresponding to the producer will not end. As a result, the life  of the batch will not end, and can not gc.
![3](https://github.com/milvus-io/milvus/assets/64083300/05c88b4e-69fe-4eba-9ae5-c8581d3e266b)
We can see if the producer is close according to the log 'produceID=XXX'
<img width="1304" alt="2" src="https://github.com/milvus-io/milvus/assets/64083300/ada42a41-0a40-498e-9486-f6e08247d1e2">
We create the replicate msg producer when start, but can not see it close.

**Comment by xiaofan-luan** [comment]
Good catch! @smellthemoon

**Comment by elstic** [comment]
This problem occurs when the mq is pulsar and the test case carries the ddl operation

**Comment by xiaofan-luan** [comment]
The main reason is due to the pulsar bufferPool.

Milvus insertion msg is large, and pulsar don't limit the memory buffer they are using.

partitionProducer -> bufferPool cost most of the memory

**Comment by elstic** [comment]
This problem still exists.

image: 2.3-20231219-cd9e3c48
case: test_concurrent_locust_100m_ivf_sq8_ddl_dql_filter_replica2_cluster 

server:
```
fouramf-fjvxb-100-4219-etcd-0                                     1/1     Running            0               14h     10.104.34.54    4am-node37   <none>           <none>
fouramf-fjvxb-100-4219-etcd-1                                     1/1     Running            0               14h     10.104.15.40    4am-node20   <none>           <none>
fouramf-fjvxb-100-4219-etcd-2                                     1/1     Running            0               14h     10.104.26.204   4am-node32   <none>           <none>
fouramf-fjvxb-100-4219-milvus-datacoord-5d67c6f8b-qljvl           1/1     Running            0               14h     10.104.29.36    4am-node35   <none>           <none>
fouramf-fjvxb-100-4219-milvus-datanode-5d8bbcfdf6-dd9km           1/1     Running            0               14h     10.104.14.23    4am-node18   <none>           <none>
fouramf-fjvxb-100-4219-milvus-indexcoord-66f475974c-qgjdp         1/1     Running            0               14h     10.104.34.44    4am-node37   <none>           <none>
fouramf-fjvxb-100-4219-milvus-indexnode-66fbd6c87b-v8vwd          1/1     Running            0               14h     10.104.14.26    4am-node18   <none>           <none>
fouramf-fjvxb-100-4219-milvus-proxy-5c7c65484f-5j7pc              1/1     Running            0               14h     10.104.14.25    4am-node18   <none>           <none>
fouramf-fjvxb-100-4219-milvus-querycoord-b887d77dd-rvg78          1/1     Running            0               14h     10.104.27.41    4am-node31   <none>           <none>
fouramf-fjvxb-100-4219-milvus-querynode-5dcd85f97b-7xcsm          1/1     Running            0               14h     10.104.29.37    4am-node35   <none>           <none>
fouramf-fjvxb-100-4219-milvus-querynode-5dcd85f97b-bp8cj          1/1     Running            0               14h     10.104.27.42    4am-node31   <none>           <none>
fouramf-fjvxb-100-4219-milvus-rootcoord-7bbc9b9f5d-s9xd8          1/1     Running            0               14h     10.104.27.37    4am-node31   <none>           <none>
fouramf-fjvxb-100-4219-minio-0                                    1/1     Running            0               14h     10.104.27.46    4am-node31   <none>           <none>
fouramf-fjvxb-100-4219-minio-1                                    1/1     Running            0               14h     10.104.25.190   4am-node30   <none>           <none>
fouramf-fjvxb-100-4219-minio-2                                    1/1     Running            0               14h     10.104.34.55    4am-node37   <none>           <none>
fouramf-fjvxb-100-4219-minio-3                                    1/1     Running            0               14h     10.104.33.2     4am-node36   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-bookie-0                            1/1     Running            0               14h     10.104.34.58    4am-node37   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-bookie-1                            1/1     Running            0               14h     10.104.25.192   4am-node30   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-bookie-2                            1/1     Running            0               14h     10.104.20.62    4am-node22   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-bookie-init-mzgld                   0/1     Completed          0               14h     10.104.27.36    4am-node31   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-broker-0                            1/1     Running            0               14h     10.104.5.55     4am-node12   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-proxy-0                             1/1     Running            0               14h     10.104.27.35    4am-node31   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-pulsar-init-hhnjf                   0/1     Completed          0               14h     10.104.27.38    4am-node31   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-recovery-0                          1/1     Running            0               14h     10.104.34.42    4am-node37   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-zookeeper-0                         1/1     Running            0               14h     10.104.27.45    4am-node31   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-zookeeper-1                         1/1     Running            0               14h     10.104.34.61    4am-node37   <none>           <none>
fouramf-fjvxb-100-4219-pulsar-zookeeper-2                         1/1     Running            0               14h     10.104.26.207   4am-node32   <none>           <none>
```

memory usage:
![image](https://github.com/milvus-io/milvus/assets/48523564/d727506b-494a-4d9b-bd5c-862141a13e3a)

**Comment by xiaofan-luan** [comment]
@smellthemoon 
we need to upgrade pulsar client and retest

**Comment by aoiasd** [comment]
Still question caused by config event but only on proxy. 
Go runtime.SetFinalizer will cause that ,if a struct has a field struct own a closure function which reference the struct(circular reference), and this struct has set finalizer by SetFinalizer, it will can't be garbage.
Milvus proxy channel manager set finalizer for all created produce stream, and mqMsgStream has configEvent which has a function will set the stream's config. Cause memory leak.

**Comment by elstic** [comment]
This fix is in effect , verify the image:image: 2.3-20231225-908e075f


After the merge of this pr [29457](https://github.com/milvus-io/milvus/pull/29457) and without the above problems, I would close it.

![image](https://github.com/milvus-io/milvus/assets/48523564/88cdf8e0-3efb-4ed5-a2b5-7c0d036be28c)

**Comment by elstic** [comment]
This problem is fixed.
